### PR TITLE
fix codelist test #750

### PR DESCRIPTION
**Pull request by btkobayashirun**
Please review #750.

### CODE REVIEW

File: terasoluna-gfw-functionaltest-web/src/main/resources/META-INF/spring/spring-mvc-common.xml [hunk 1]
```diff
@@ -79,31 +79,19 @@
...
     <mvc:interceptor>
       <mvc:mapping path="/codelist/fallbackToFr/**" />
       <mvc:exclude-mapping path="/resources/**" />
       <mvc:exclude-mapping path="/**/*.html" />
       <bean class="org.terasoluna.gfw.web.codelist.CodeListInterceptor">
-        <property name="codeListIdPattern" value="CL_.+" />
-        <property name="fallbackTo" value="fr" />
+        <property name="codeListIdPattern" value="CLFR_.+" />
       </bean>
     </mvc:interceptor>
```
Comment (yoshikawaa) [inline terasoluna-gfw-functionaltest-web/src/main/resources/META-INF/spring/spring-mvc-common.xml]: - [ ] You no longer need to use `CodeListInterceptor` differently to change the fallback locale. Do you need to define multiple fallback `CodeListInterceptors`?

File: terasoluna-gfw-functionaltest-domain/src/main/resources/META-INF/spring/terasoluna-gfw-functionaltest-codelist.xml [hunk 2]
```diff
@@ -72,7 +72,7 @@
   </bean>
 
   <bean id="CL_CODELIST06_01" class="org.terasoluna.gfw.common.codelist.i18n.SimpleI18nCodeList">
-    <property name="rows"> <!-- (1) -->
+    <property name="rows">
       <util:map>
         <entry key="en">
```
[the remark this answers is not in the document]
Comment (btkobayashirun) [reply terasoluna-gfw-functionaltest-domain/src/main/resources/META-INF/spring/terasoluna-gfw-functionaltest-codelist.xml]: Modified by 8ec78f399640dc8290f1d8b761eb9aac2a181f9d
`fallbackTo` specified en.

File: terasoluna-gfw-functionaltest-domain/src/main/resources/META-INF/spring/terasoluna-gfw-functionaltest-codelist.xml [hunk 3]
```diff
@@ -92,28 +92,36 @@
...
+        <entry key="key3" value="label3" />
+      </util:map>
+    </property>
   </bean>
 
-  <bean id="CL_CODELIST06_06_JA" parent="AbstractJdbcCodeList">
-    <property name="querySql" value="select code, label_ja from item3 ORDER BY code" />
-    <property name="valueColumn" value="code" />
-    <property name="labelColumn" value="label_ja" />
+  <bean id="CL_CODELIST06_02_JA" class="org.terasoluna.gfw.common.codelist.SimpleMapCodeList">
```
[the remark this answers is not in the document]
Comment (btkobayashirun) [reply terasoluna-gfw-functionaltest-domain/src/main/resources/META-INF/spring/terasoluna-gfw-functionaltest-codelist.xml]: change locale to`ja`.
The change is carried out collectively with the following commit.
8ec78f399640dc8290f1d8b761eb9aac2a181f9d

File: terasoluna-gfw-functionaltest-web/src/main/resources/META-INF/spring/spring-mvc-common.xml (hunk 1 above)
Comment (btkobayashirun) [reply terasoluna-gfw-functionaltest-web/src/main/resources/META-INF/spring/spring-mvc-common.xml]: Modified by ec1c229ced6a7858f02e35d02e352e85eac81f59
I did not need it, so I erased it.

**Comment by btkobayashirun** [comment]
The contents changed in the review of the wiki are reflected here.
befac2ab17044c7bfcf71a7d761595eccbd97a80

File: terasoluna-gfw-functionaltest-domain/src/main/resources/META-INF/spring/terasoluna-gfw-functionaltest-codelist.xml [hunk 4]
```diff
@@ -90,54 +90,100 @@
...
-            <entry key="ja_JP" value="ラベル3" />
+            <entry key="ja" value="ラベル3" />
           </util:map>
         </entry>
       </util:map>
     </property>
+    <property name="fallbackTo" value="en" />
   </bean>
 
+  <bean id="CL_CODELIST11_01" class="org.terasoluna.gfw.common.codelist.i18n.SimpleI18nCodeList">
```
[the remark this answers is not in the document]
Comment (btkobayashirun) [reply terasoluna-gfw-functionaltest-domain/src/main/resources/META-INF/spring/terasoluna-gfw-functionaltest-codelist.xml]: We added CL_CODELIST11_05.
5184861b15a79e39fbc4fbbe5517e41e68508c97
Therefore, we have not made this correspondence

File: terasoluna-gfw-functionaltest-web/src/main/webapp/WEB-INF/views/codelist/index.jsp [hunk 5]
```diff
@@ -2,142 +2,125 @@
 
 <label>Change Locale Link</label>
 <a href="${pageContext.request.contextPath}/codelist?locale=en">English</a>
-<a href="${pageContext.request.contextPath}/codelist?locale=ja_JP">Japanese</a>
-<a href="${pageContext.request.contextPath}/codelist?locale=fr">French</a>
-<a href="${pageContext.request.contextPath}/codelist?locale=cn">Chinese</a>
+<a href="${pageContext.request.contextPath}/codelist?locale=fr_fr">Franc</a>
```
[the remark this answers is not in the document]
Comment (btkobayashirun) [reply terasoluna-gfw-functionaltest-web/src/main/webapp/WEB-INF/views/codelist/index.jsp]: Modified by 4ae91c01540a4910c3449ac3419b886e1ec596c5

**Comment by btkobayashirun** [comment]
Test added by wiki modification.
Skips when `assumeThat` does not satisfy prerequisites `ja_JP`.
 5184861b15a79e39fbc4fbbe5517e41e68508c97

File: terasoluna-gfw-functionaltest-selenium/src/test/java/org/terasoluna/gfw/functionaltest/app/codelist/CodeListTest.java [hunk 6]
```diff
@@ -594,4 +597,20 @@ public void test11_04_form() {
...
+        driver.findElement(By.id("codelist_11_05")).click();
+
+        assumeThat(driver.findElement(By.id("defaultLocale")), is(
+                "The default locale is " + Locale.JAPAN));
+        assertThat(driver.findElement(By.cssSelector("option[value=\"key1\"]"))
+                .getText(), is("ラベル1"));
+        assertThat(driver.findElement(By.cssSelector("option[value=\"key2\"]"))
+                .getText(), is("ラベル2"));
+        assertThat(driver.findElement(By.cssSelector("option[value=\"key3\"]"))
+                .getText(), is("ラベル3"));
```
Comment (yoshikawaa) [inline terasoluna-gfw-functionaltest-selenium/src/test/java/org/terasoluna/gfw/functionaltest/app/codelist/CodeListTest.java]: - [ ] Assuming a bean definition, You can make succeed test when default locale is `ja`, `ja_JP`, `en`, `en_US` etc...
You should change to apply this. ex...

* Controller
Set default language locale.
```java
Locale defaultLocale = Locale.getDefault();
if (defaultLocale.getCountroy().length > 0) {
  defaultLocale = new Locale(defaultLocale.getLanguage());
}
model.addAttribute("defaultLocale", defaultLocale);
```
* JUnit
Assume when default locale is `jp` or `en`.
```java
String textDefaultLocale = driver.findElement(By.id("defaultLocale")).getText();
assumeThat(textDefaultLocale ,
  anyOf(containsString(Locale.JAPANESE), containsString(Locale.ENGLISH));
if (textDefaultLocale.contains(Locale.JAPANESE) {
  // assert Japanese labels.
} else {
  // assert English labels.
}
```

It seems also good to remove `en` codelist from `CL_CODELIST11_05` and make it simple.

Comment (btkobayashirun) [reply terasoluna-gfw-functionaltest-selenium/src/test/java/org/terasoluna/gfw/functionaltest/app/codelist/CodeListTest.java]: I excluded the implementation related to `test11_05_form` and gave `@Ignore` for the test.
a6c96bda709fdf5f46917c7ca0efd9ebec822f73

Note : There is a possibility that application startup may fail if implemented.
　　　Also, we implemented unit test of this function with terasoluna-gfw.
　　　Therefore, correspondence was made.

**Pull request merged**